File: cache/src/main/java/io/github/manami/cache/strategies/headlessbrowser/extractor/relatedanime/RelatedAnimeExtractor.java
package io.github.manami.cache.strategies.headlessbrowser.extractor.relatedanime;

import io.github.manami.cache.strategies.headlessbrowser.extractor.AnimeExtractor;
import io.github.manamiproject.manami.dto.entities.InfoLink;
import java.util.Set;

/**
 * Can extract links to related anime.
 */
public interface RelatedAnimeExtractor extends AnimeExtractor {

  /**
   * Extracts links to related anime.
   *
   * @return A list of URLs for related anime.
   */
  Set<InfoLink> extractRelatedAnime(String siteContent);
}

File: cache/src/main/kotlin/io/github/manamiproject/manami/cache/remoteretrieval/extractor/relatedanime/RelatedAnimeExtractor.kt
package io.github.manamiproject.manami.cache.remoteretrieval.extractor.relatedanime

import io.github.manamiproject.manami.cache.remoteretrieval.extractor.Extractor
import io.github.manamiproject.manami.entities.InfoLink

/**
 * Extracts a list of related anime from a source. A raw HTML page for example.
 */
internal interface RelatedAnimeExtractor : Extractor {

    /**
     * Extracts the list of related anime from a source String.
     * @param source Any given source. A raw HTML page for example.
     * @return A Set of directly related anime represented with their InfoLinks.
     */
    fun extractRelatedAnime(source: String): Set<InfoLink>
}